pandas/core/frame.py

```diff
@@ -9652,6 +9653,18 @@ def apply(
         args : tuple
             Positional arguments to pass to `func` in addition to the
             array/series.
+        by_row : bool or "compat", default "compat"
+            If "compat", will if possible first translate the func into pandas
+            methods (e.g. ``Series().apply(np.sum)`` will be translated to
+            ``Series().sum()``). If that doesn't work, will try call to apply again with
+            ``by_row=True`` and if that fails, will call apply again with
+            ``by_row=False``
+            If True, gives the same results as for "compat".
```

Comment:
Is this in all cases? If that's the case, I'm not sure why we need it.

Reply:
Friendly ping here @topper-123

Reply:
Sorry about the slow reply, I had a project I had to finish before going on a family vacation, which I'm on currently.

For `Series.apply`, `by_row` we needs the 3 possible values. I gave `DataFrame.apply` the same signature for `by_row`, though `by_row=True` wasn't implemented for `DataFrame.apply` just so the signatures would be the same.

An alternative would be to implement `by_row=True` for `DataFrame.apply`, but if the intention is to deprecate `by_row="compat"` and `by_row=True`, I thought it wasn't worth is to add that functionality.

Just to be clear, my intention is to deprecate `by_row="compat"` and `by_row=True`, though I probably will wait till after v2.1 has been released to do that.

Reply:
No worries! Just wanted to be sure the question was seen.

I'm still finding myself a little confused here - in previous versions of pandas the apply and/or agg operation would either act as `by_row=True` or `by_row=False` depending on what seems to be somewhat difficult to describe conditions (not that we can't - just that it's quite a bit of effort to do so accurately). It seems like here users may specify `by_row=True` and then we may (or we are always?) switching that to `by_row="compat"`. This seems confusing from a user perspective if they specify `by_row=True` but instead get `by_row=False`-like behavior.

In 3.0, we really just want `by_row=False` for both agg and apply. Assuming what I said above is right, what if we just allowed `by_row` to be `False` or `"compt"` but not `True`?

Reply:
> Assuming what I said above is right, what if we just allowed by_row to be False or "compt" but not True?

Yeah, that’s a better idea. I’ll update when I get a moment to do it..

Reply:
I've updated.